apps/guide/src/content/02-creating-your-bot/03-command-handling.mdx

```diff
@@ -0,0 +1,220 @@
+client.on(Events.InteractionCreate, (interaction) => {
+	if (!interaction.isChatInputCommand()) return;
+	console.log(interaction);
+});
+```
+
+</CH.Code>
+
+## Executing commands
+
+When your bot receives a <DocsLink type="class" parent="Client" symbol="e-interactionCreate" /> event, the interaction object contains all the information you need to dynamically retrieve and execute your commands!
+
+Let's take a look at the _`ping`_ command again. Note the _`execute()`_ function that will reply to the interaction with "Pong!".
+
+<CH.Code>
+
+```js ping.js
+module.exports = {
+	data: new SlashCommandBuilder().setName('ping').setDescription('Replies with Pong!'),
+	async execute(interaction) {
+		await interaction.reply('Pong!');
+	},
+};
+```
+
+</CH.Code>
+
+First, you need to get the matching command from the _`client.commands`_ Collection based on the _`interaction.commandName`_. Your <DocsLink type="class" parent="Client"/> instance is always available via _`interaction.client`_. If no matching command is found, log an error to the console and ignore the event.
+
+With the right command identified, all that's left to do is call the command's _`.execute()`_ method and pass in the _`interaction`_ variable as its argument. Note that the event listener has been made _`async`_, allowing Promises to be awaited. In case something goes wrong and the Promise rejects, catch and log any error to the console.
+
+<CH.Code>
+
+```js index.js focus=4:20
+// focus[37:42]
+client.on(Events.InteractionCreate, async (interaction) => {
+	if (!interaction.isChatInputCommand()) return;
+
+	const command = interaction.client.commands.get(interaction.commandName);
+
+	if (!command) {
+		console.error(`No command matching ${interaction.commandName} was found.`);
+		return;
+	}
+
+	try {
+		await command.execute(interaction);
+	} catch (error) {
+		console.error(error);
+		if (interaction.replied || interaction.deferred) {
+			await interaction.followUp({ content: 'There was an error while executing this command!', ephemeral: true });
+		} else {
+			await interaction.reply({ content: 'There was an error while executing this command!', ephemeral: true });
+		}
+	}
+});
+```
+
+</CH.Code>
+
+## Command categories
+
+So far, all of your command files are in a single _`commands`_ folder. This is fine at first, but as your project grows, the number of files in the _`commands`_ folder will too. Keeping track of that many files can be a little tough. To make this a little easier, you can categorize your commands and put them in subfolders inside the _`commands`_ folder. You will have to make a few changes to your existing code in _`index.js`_ for this to work out.
+
+If you've been following along, your project structure should look something like this:
+
+![Project structure before sorting](./images/before-sorting.png)
+
+After moving your commands into subfolders, it will look something like this:
+
+![Project structure after sorting](./images/after-sorting.png)
```

Comment:
Right now, I've put all guide-related images in /public/assets. If you want to move these over now (or take updated ones), up to you.